test/controllers/liking_relations_controller_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+class LikingRelationsNotLoggedIn < ActionDispatch::IntegrationTest
+
+  def setup
+    @relation = liking_relations(:one_likes_two)
+  end
+end
+
+class LikingRelationsNotLoggedInTest < LikingRelationsNotLoggedIn
+
+  test 'should redirect create if not logged in' do
+    post liking_relations_path
+    assert_redirected_to login_url
+    assert_response :see_other
+    follow_redirect!
+    assert_select 'div.alert-warning'
```

Comment:
generalnie takie testy mogą też ci kiedyś dać false negative - jeśli przekierowałoby Cię na jakąś stronę z alertem "Uwaga - promocja z okazji black friday" to by test przeszedł, mimo iż oczekiwałeś alerta "Musisz być zalogowany" czy coś takiego.

Można dodatkowo sprawdzić tekst alertu chyba?

Reply:
Tak coś czułem, że to niedokładne, ale jakoś olałem. Masz rację. Poprawiam.